File: DivisibleSetDiv2/solve/ruby/DivisibleSetDiv2Solver.rb
#! /usr/bin/env ruby

require_relative "DivisibleSetDiv2"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  b = reader.next(TArray.new(TInt))

  result = DivisibleSetDiv2.new().isPossible(b)
  IO.write(ARGV[1], Writer.new.next(result, TString).to_s)
end

init
main

File: DivisibleSetDiv2/solve/python/DivisibleSetDiv2Solver.py
#! /usr/bin/env python
import os, sys
import DivisibleSetDiv2

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)

        b = reader.next("int[]")

    result = DivisibleSetDiv2.isPossible(b)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "String"))

if __name__ == "__main__":
    init()
    main()
